compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -276,18 +276,19 @@ static LogicalResult setContractConfig(func::FuncOp entryPoint,
 
   // Also exclude the case of matvec, which has only one non-unit parallel dim.
   // They should go down different pipelines.
-  int nonUnitParallelDimCount = 0;
+  // Currently dynamic dimensions are treated like unit dimensions in codegen.
+  int staticNonUnitParallelDimCount = 0;
   SmallVector<int64_t, 4> bounds = op.getStaticLoopRanges();
   FailureOr<mlir::linalg::ContractionDimensions> contractionDims =
       mlir::linalg::inferContractionDims(op);
   assert(succeeded(contractionDims) && "Could not infer contraction dims");
   for (auto mDim : contractionDims->m) {
-    nonUnitParallelDimCount += bounds[mDim] != 1;
+    staticNonUnitParallelDimCount += bounds[mDim] != 1 && bounds[mDim] != ShapedType::kDynamic;
   }
   for (auto nDim : contractionDims->n) {
-    nonUnitParallelDimCount += bounds[nDim] != 1;
+    staticNonUnitParallelDimCount += bounds[nDim] != 1 && bounds[nDim] != ShapedType::kDynamic;
```

Comment:
nit: use `ShapedType::isDynamic(M)`. I recently learned that one of MLIR goals is to make ShapedType::kDynamic private, and force all users to go through safe APIs.

(just in case, https://discourse.llvm.org/t/rfc-unify-kdynamicsize-and-kdynamicstrideoroffset/64534 has a little context. What I learned is more from review comments: https://reviews.llvm.org/D138282)